eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -243,11 +280,10 @@ jobs:
               ${{ insert }}: ${{ matrixEntry.value }}
 
     pool:
-      vmImage: $(OSVmImage)
+      name: $[coalesce(variables['Pool'], '')]
+      vmImage: $[coalesce(variables['OSVmImage'], '')]
 
     steps:
-      - template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
```

Comment:
I would really like us to have some support for verifying the agent os.

Reply:
It seems unnecessary to me under the new pool model.

Reply:
I feel it is even more necessary under the new pool model, at least until we have confidence that we are running against the correct in each pool. My biggest concern is we configure a pool wrong by point at an incorrect image, or even if somewhere in the 1ES system the wrong image gets picked we may not find out for a long time that we have been running on the wrong OS.